ui/selectors/assets.ts

```diff
@@ -210,3 +211,65 @@ export const getMultiChainAssets = createDeepEqualSelector(
     });
   },
 );
+
+export const getMultichainAggregatedBalance = createDeepEqualSelector(
+  (_state, selectedAccount) => selectedAccount,
+  getMultichainNetwork,
+  getMultichainBalances,
+  getAccountAssets,
+  getAssetsRates,
+  (
+    selectedAccountAddress,
+    currentNetwork,
+    multichainBalances,
+    accountAssets,
+    assetRates,
+  ) => {
+    const assetIds = accountAssets?.[selectedAccountAddress.id] || [];
+    const balances = multichainBalances?.[selectedAccountAddress.id];
+
+    let aggregatedBalance = new BigNumber(0);
+
+    assetIds.forEach((assetId: CaipAssetId) => {
+      const { chainId } = parseCaipAssetType(assetId);
+      if (chainId === currentNetwork.chainId) {
+        const balance = balances?.[assetId] || { amount: '0', unit: '' };
+        const rate = assetRates?.[assetId]?.rate || '0';
+        const balanceInFiat = new BigNumber(balance.amount).times(rate);
+
+        aggregatedBalance = aggregatedBalance.plus(balanceInFiat);
+      }
+    });
+
+    return aggregatedBalance.toNumber();
+  },
+);
+
+export const getMultichainNativeTokenBalance = createDeepEqualSelector(
+  (_state, selectedAccount) => selectedAccount,
+  getMultichainNetwork,
+  getMultichainBalances,
+  getAccountAssets,
+  (
+    selectedAccountAddress,
+    currentNetwork,
+    multichainBalances,
+    accountAssets,
+  ) => {
+    const assetIds = accountAssets?.[selectedAccountAddress.id] || [];
+    const balances = multichainBalances?.[selectedAccountAddress.id];
+
+    let nativeTokenBalance = { amount: 0, unit: '' };
```

Comment:
Could we turn this into a selector so that it returns the same object from `createDeepEqualSelector`?  Otherwise this is a new object which wouldn't be memoized.